libraries/IiifItems/Util/Canvas.php

```diff
@@ -169,7 +169,12 @@ public static function fileImageJson($file, $on, $force=false) {
         catch (Exception $e) {
         }
         // If missing or failed, build from file data
-        list($fileWidth, $fileHeight) = getimagesize(FILES_DIR . DIRECTORY_SEPARATOR . $file->getStoragePath());
+        // Commented code relies on file being in files dir, which is not the case if using a storage adapter
+        // Implemented solution uses
+        // list($fileWidth, $fileHeight) = getimagesize(FILES_DIR . DIRECTORY_SEPARATOR . $file->getStoragePath());
+        $fileMetadata = json_decode($file->metadata);
```

Comment:
Note that [json_decode() will return NULL](https://www.php.net/manual/en/function.json-decode.php#refsect1-function.json-decode-returnvalues) if the metadata is empty/invalid, so it would be worth checking. I think something like this would at least catch it and log it for further investigation (untested):

```
if($fileMetadata === NULL || !isset($fileMetadata->video)) {
    _log("Cannot get image size metadata for file {$file->id}", Zend_Log::WARN);
    $fileWidth = 0;
    $fileHeight = 0;
} else {
    $fileWidth = $fileMetadata->video->resolution_x;
    $fileHeight = $fileMetadata->video->resolution_y;
}
```